src/Engine/ProtoCore/Namespace/SymbolTable.cs

```diff
@@ -103,6 +242,28 @@ public override bool Equals(object obj)
         }
 
         /// <summary>
+        /// Equality operator overload to have same equality behaviour as object.Equals() override
+        /// </summary>
+        /// <param name="thisSymbol"></param>
+        /// <param name="otherSymbol"></param>
+        /// <returns></returns>
+        public static bool operator ==(Symbol thisSymbol, Symbol otherSymbol)
```

Comment:
was there a specific reason to use an overload op as opposed to a normal function name?